### DISCUSSION TITLE
How to handle potential multiple alerts by super-scene and sub-scene?

### DISCUSSION THREAD

**Discussion by ghost** [pointfreeco/swift-composable-architecture#2232] [Q&A]
Hello :)

I have the situation where scene A _contains_ (ie, this is not a presentation) scene B and both scenes have state, actions, and reducers of their own. Moreover, both scenes can show alerts to the user and therein lies a potential conflict where we might have two alerts showing at the same time.

It seems to me that having either scene pass information about their current `destination` to the other is a break in encapsulation. It shouldn't be either scene's responsibility to know or care about what alerts the other might be showing when it wants to show an alert of its own.

How would you handle such a scenario?

Many thanks in advance for any recommendations.

**Comment by mbrandonw** [comment]
Hi @wtruppel, the only idea that comes to mind is for scene B to communicate to the parent when it wants an alert shown rather than showing the alerts itself. Probably using delegate actions.

But also it may be worth weighing the tradeoffs of a fully concise domain and one that is maybe less than ideal. Is the uncertainty of having two features showing an alert at the same time worse than the work it takes to solve the problem? If not, then maybe it is ok to use the less concisely modeled domain until it definitely becomes a problem.

**Reply by ghost** [reply]
Hi @mbrandonw, thanks for the quick reply.

Yeah, I've been leaning towards the delegate action pattern. It seems to me to be the solution with the least amount of friction. I thought I'd ask anyway just in case people had better suggestions. Thanks again.
